src/ddmd/func.d

```diff
@@ -3189,6 +3182,30 @@ extern (C++) final class UnitTestDeclaration : FuncDeclaration
         return FuncDeclaration.syntaxCopy(utd);
     }
 
+    final void setIdentifier()
+    {
+        ident = createIdentifier(loc, _scope);
+    }
+
+    /***********************************************************
+     * Generate unique unittest function Id so we can have multiple
+     * instances per module.
+     */
+    private static Identifier createIdentifier(Loc loc, Scope* sc)
+    {
+
+        OutBuffer buf;
+        auto index = sc ? sc._module.unitTestCounter++ : 0;
+        buf.printf("__unittest_%s_%u_%d", loc.filename, loc.linnum, index);
+
+        // replace characters that demangle can't handle
+        auto str = buf.peekString;
+        for(int i = 0; str[i] != 0; ++i)
+            if(str[i] == '/' || str[i] == '\\' || str[i] == '.') str[i] = '_';
```

Comment:
More efficient as:
```
foreach (ref c; buf.peekSlice())
{
    if (c == '/' || c == ...) c = '_';
}
```

Reply:
Or even: `buf.peekSlice()[11 .. $-4]` :-)

Reply:
`peekSlice` returns `const(char)[]` so that doesn't compile. The latter version is also too hardcoded, even though I think it's unlikely anybody will change the way unit tests are named in the future.

Reply:
here's one of those places where `among` is useful (though we can't use it here)

Reply:
> so that doesn't compile

Oopsie-daisy!

Reply:
This test is not strong enough and various testsuite tests fail if compiling to assembly (such as what gdc does).

This logic should match what is done for `pragma(mangle)`.

https://github.com/dlang/dmd/blob/32830defad38c5dec678c13bed105d26b8c4ee12/src/ddmd/dsymbolsem.d#L3033-L3045

Reply:
For instance, unittests compiled that are inside the directory `extra-files` fail because the assembler stumbles over on the dash symbol.